examples/foundational/14-function-calling.py

```diff
@@ -34,7 +34,12 @@
 
 
 async def start_fetch_weather(function_name, llm, context):
-    await llm.push_frame(TextFrame("Let me check on that."))
+    # note: we can't push a frame to the LLM here. the bot
+    # can interrupt itself and/or cause audio overlapping glitches.
+    # possible question for Aleix and Chad about what the right way
+    # to trigger speech is, now, with the new queues/async/sync refactors.
+    # await llm.push_frame(TextFrame("Let me check on that."))
+    logger.debug(f"Starting fetch_weather_from_api with function_name: {function_name}")
```

Comment:
The rule is... if a processor can push frames from a different task then it should be async. But function calls happen in the same task, no? If so, this should be OK.

Reply:
I just realized that, by default, all processors should be async and not the other way around. It makes more sense and it's safer this way. That was my initial idea but then changed my mine. But I think that was the right idea. I'll make the change in the morning and I believe that line should be safe.

Reply:
I will rebase and test again after this change.